api/restHandler/PipelineConfigRestHandler.go

```diff
@@ -604,6 +606,14 @@ func (handler PipelineConfigRestHandlerImpl) ConfigureDeploymentTemplateForApp(w
 		writeJsonResp(w, err, nil, http.StatusBadRequest)
 		return
 	}
+	ChartVersion := templateRequest.RefChartTemplate
+
+	validate, error := validator2.DeploymentTemplateValidate(templateRequest.ValuesOverride, ChartVersion)
```

Comment:
please move this step after validating struct and RBAC

Reply:
please combine this service with create service

Reply:
invoke this method as singleton